tests/test_nms.cpp

```diff
@@ -57,3 +57,148 @@ TEST(TestSoftmax, Hello) {
   float output = softmax(input_vec);
   ASSERT_EQ(req_out, output);
 }
+
+TEST(TestGetNewOrder, Hello) {
+  std::vector<int> input_order{5, 4, 2, 1, 89};
+  std::vector<int> indices{0, 1, 2, 3};
+  std::vector<int> req_out{2, 1};
+
+  get_new_order(input_order, indices);
+  ASSERT_EQ(req_out, input_order);
+}
+
+bool floatCompare(float f1, float f2) {
+  static constexpr auto epsilon = 1.0e-01f;
+  if (fabs(f1 - f2) <= epsilon)
+    return true;
+  return false;
+}
+
+TEST(TestDistances, Hello) {
+  AnchorMat test_input = create_sample_input();
+
+  std::vector<int> x_indices = create_sequence_vector(
+      x_coordinate_start_idx, x_coordinate_start_idx + predication_steps - 1);
+  std::vector<int> z_indices = create_sequence_vector(
+      z_coordinate_start_idx, z_coordinate_start_idx + predication_steps - 1);
+
+  std::vector<std::vector<float>> all_x =
+      extract_columns<proposal_dim>(test_input, x_indices);
+  std::vector<std::vector<float>> all_z =
+      extract_columns<proposal_dim>(test_input, z_indices);
+
+  std::vector<float> expected_output = {0,        0.141421, 0.282843,
+                                        0.424264, 0.565685, 0.707107};
+  std::vector<float> accumulated_dis =
+      compute_all_lane_distance(all_x, all_z, all_x[0], all_z[0]);
+  for (int i = 0; i < accumulated_dis.size(); ++i) {
+    ASSERT_TRUE(floatCompare(accumulated_dis[i], expected_output[i]));
+  }
+  // ASSERT_EQ(req_out, input_order);
+}
+
+AnchorMat conf_testing() {
+  AnchorMat anchor_vec;
+
+  Anchor anchor_2{0.7, 1.3, 1.4, 0.6, 0.6};
```

Comment:
Please add the motivation of why this test case was chosen and why its important

Reply:
Done!